src/server.c

```diff
@@ -1276,6 +1277,25 @@ void qd_server_set_container(qd_dispatch_t *qd, qd_container_t *container)
     qd->server->container = container;
 }
 
+void qd_server_trace_all_connections()
+{
+    qd_dispatch_t *qd = qd_dispatch_get_dispatch();
+    if (qd->server) {
+        qd_connection_list_t  conn_list = qd->server->conn_list;
+        qd_connection_t *conn = DEQ_HEAD(conn_list);
+        while(conn) {
```

Comment:
the qd->server->conn_list is protected by qd->server->lock.  this loop will need to hold that lock while walking the list or it is possible the connection it is working on is removed (and possibly freed) by another thread.